packages/intrepid2/src/Discretization/Basis/Intrepid2_HGRAD_QUAD_C1_FEMDef.hpp

```diff
@@ -161,7 +161,7 @@ namespace Intrepid2 {
                const EOperator operatorType )  {
       typedef          Kokkos::DynRankView<outputValueValueType,outputValueProperties...>         outputValueViewType;
       typedef          Kokkos::DynRankView<inputPointValueType, inputPointProperties...>          inputPointViewType;
-      typedef typename ExecSpace<typename inputPointViewType::execution_space,typename DT::execution_space>::ExecSpaceType ExecSpaceType;
+      typedef typename ExecSpace<typename inputPointViewType::execution_space,SpT>::ExecSpaceType ExecSpaceType;
```

Comment:
@CamelliaDPG You overwrote what I changed in the previous code. As all Intrepid2 class is planned to be with device type, this SpT is not right. It looks like that you run a sed script but I do not think that it is safe.

Reply:
@kyungjoo-kim, sorry about that; I've now restored what you had.  I didn't intend to overwrite your changes.  I changed course in terms of the way I was handling the Impl:: classes midway through.  I had thought that we could make the DeviceType change in the outward-facing classes and still use ExecutionSpace for the Impl, but perhaps that's incorrect.  (I was working quickly, but not using scripts.)

Reply:
> @CamelliaDPG This does not seem to be properly tested without uvm and some parts of the code is overwritten incorrectly. Could we change the code incrementally from basis ?

@kyungjoo-kim, my basic approach here was to make changes in the various structured integration classes.  This led to needing to make changes in many other places just to get things building; it's close to as incremental as I could make it, starting from where I did.  Part of my effort to keep it incremental, especially in light of your concerns about slowing down tests, was not to include the UVM-free tests; those can clearly be done in an incremental way, one at a time.  I did quite a bit of work here, and would be sorry to throw it away, as you seem to be suggesting.

So, changes that should be made to this PR:
- I can change the Basis `Impl` classes to use DeviceType.
- You also recommended against using a default template argument of `Kokkos::DefaultExecutionSpace::device_type`, but you didn't say why you recommend that, and I have some reasons for wanting it.  Would you explain what your concern is?
- Any other changes to the source (as opposed to the tests) that you think need to be made?

Reply:
- To test UVM free, we are using non-default device type. By defaulting the template argument, some code path and test can use a different device including uvm memory space. 
- I do not think it is good to merge partially completed code wihtout unit testing. For many classes included in this PR, they have unit tests. Testing the unit tests without UVM is the necessary condition for this work as its objective is removal of UVM. 
- I see that this PR is big but it is "not" tested.

Reply:
@kyungjoo-kim, thank you for clarifying.

I understand why you recommend not to default the template argument; I prefer the default because it's the case most users will want, and including it allows user code to be more succinct.

I will defer to your rules regarding testing.  I thought that we could regard this PR as not introducing a regression supposing the tests passed (since the tests would cover existing functionality), and it counts as progress, since it puts us in a position to write the UVM-free tests.  But I can see that's not your interpretation.